### PR TITLE
Add documentation for Tekton support

### PR DESCRIPTION
**Pull request by flamywhale**
Add documentation for Tekton support including usage , design and implementation details

### CODE REVIEW

File: docs/tekton-support.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+## Quick start
+
+The design of Tekton integration is currently on the branch [tekton-support](https://github.com/kubesphere/ks-devops/tree/tekton-support) under development.
+
+If you want to try it out, you can follow the below instructions.
+
+1. Change the branch to [tekton-support](https://github.com/kubesphere/ks-devops/tree/tekton-support).
+    * `git checkout tekton-support`
+1. Setup a clean kubernetes cluster (v1.19.8 is **recommended**) using [kind](https://kind.sigs.k8s.io/), [kubectl](https://kubernetes.io/docs/reference/kubectl/overview/) or [kk](https://github.com/kubesphere/kubekey).
```
Comment (LinuxSuRen) [inline docs/tekton-support.md:26]: `kubernetes` -> `Kubernetes`

File: docs/tekton-support.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    * `uses` provides the template task name.
+    * `with` provides the params needed to pass to the template.
+
+### Controllers implementation
+
+#### Pipeline controller
+
+* `Pipeline` controller is located at `controllers/tekton/pipeline/pipeline_controller.go`.
+* The controller is registered at `cmd/controller/app/controllers.go` in `addControllers` function.
+* The core logic of reconcile Pipeline crd resources is as follows:
```
Comment (LinuxSuRen) [inline docs/tekton-support.md:178]: `crd` -> `CRD`

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
Thanks for your great document. But there are some typos you need to take a look it.

File: docs/tekton-support.md (hunk 1 above)
Comment (flamywhale) [reply docs/tekton-support.md:26]: OK.

File: docs/tekton-support.md (hunk 2 above)
Comment (flamywhale) [reply docs/tekton-support.md:178]: Okay.

**Review by LinuxSuRen — COMMENTED** [review]
/lgtm

**Pull request merged**